### PR TITLE
[RA2 Ch2] CPU-related language corrections

### LINKED ISSUE #2036: [RA2 Ch2] CPU-related language corrections
Terminology and language corrections are needed to ensure consistent usage of process, vCPU, CPU, and physical CPU.

### PR DESCRIPTION
**Pull request by EditKoselak**
closes #2036

### CODE REVIEW

**Review by iangardner22 — APPROVED** [review]

**Comment by pgoyal01** [comment]
@EditKoselak Doesn't Kubernetes use portions of the physical core (in thousandths of a core not a thread) including the threads? Isn't the command for the reservation:
 --kube-reserved=[cpu=2000m]  can also be  --kube-reserved=[cpu=2]  
and this would reserve 2 complete cores.

**Comment by tomkivlin** [comment]
> @EditKoselak Doesn't Kubernetes use portions of the physical core (in thousandths of a core not a thread) including the threads? Isn't the command for the reservation:
> --kube-reserved=[cpu=2000m] can also be --kube-reserved=[cpu=2]
> and this would reserve 2 complete cores.

Depends on whether the node is a VM or physical.  If physical then yes - it is reserving physical cores.  If it's a VM then it is reserving vCPUs.

**Review by petorre — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Comment by tomkivlin** [comment]
@pgoyal01 - any further thoughts on this following my reply?  @EditKoselak - can you address Pankaj's concerns please?

> Doesn't Kubernetes use portions of the physical core (in thousandths of a core not a thread) including the threads? Isn't the command for the reservation:
--kube-reserved=[cpu=2000m] can also be --kube-reserved=[cpu=2]
and this would reserve 2 complete cores.

**Review by ulikleber — APPROVED** [review]

**Comment by CsatariGergely** [comment]
> 
> 
> > @EditKoselak Doesn't Kubernetes use portions of the physical core (in thousandths of a core not a thread) including the threads? Isn't the command for the reservation:
> > --kube-reserved=[cpu=2000m] can also be --kube-reserved=[cpu=2]
> > and this would reserve 2 complete cores.
> 
> Depends on whether the node is a VM or physical. If physical then yes - it is reserving physical cores. If it's a VM then it is reserving vCPUs.
> 
> @tomkivlin Thanks. Then the language would need to indicate that. The language add about SMT doesn't do so.

@tomkivlin  For the infrastructure is is just a CPU core regardless if it is virtualized of physical. I would just keep this as "CPU".

**Comment by EditKoselak** [comment]
@tomkivlin, are you OK with the solution proposed by @CsatariGergely?

**Comment by tomkivlin** [comment]
> @tomkivlin, are you OK with the solution proposed by @CsatariGergely?

Yes that works for me. @pgoyal01 ?

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ To ensure alignment with the infrastructure profile catalogue, the following req
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.015 | SmartNICs | Not required | Optional | |
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.016 | FPGA/other Acceleration H/W | Not required | Optional | |
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | *e.cap.017* | *Ability to monitor L2-L7 data from workload* | *n/a<sup>(3)</sup>* | *n/a<sup>(3)</sup>* | |
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Indicates the number of CPU cores consumed by the Cloud Infrastructure on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | |
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Specifies the proportion of CPU cores reserved for the Cloud Infrastructure system on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:65]: In RM 4.2.5, i.cap.014 is defined as consumed not reserved:

i.cap.014(CPU cores consumed by the Cloud Infrastructure on the worker nodes) | any | any
-- | -- | --

```suggestion
| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Specifies the proportion of CPU coresconsumed by the Cloud Infrastructure system on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | |
```

**Review by pgoyal01 — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ To ensure alignment with the infrastructure profile catalogue, the following req
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.015 | SmartNICs | Not required | Optional | |
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.016 | FPGA/other Acceleration H/W | Not required | Optional | |
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | *e.cap.017* | *Ability to monitor L2-L7 data from workload* | *n/a<sup>(3)</sup>* | *n/a<sup>(3)</sup>* | |
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Indicates the number of CPU cores consumed by the Cloud Infrastructure on the worker nodes | 2 | 2 | |
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Specifies the proportion of CPU cores consumed by the Cloud Infrastructure system on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | |
```
Comment (ulikleber) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:65]: The wording needs some clarification:
While for CPU cores it says "proportion" without saying proportion to what, while for SMT threads it says absolute number.

**Review by tomkivlin — APPROVED** [review]

**Pull request merged**